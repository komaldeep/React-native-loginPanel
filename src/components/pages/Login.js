import React, { PureComponent } from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import multipleStyles from 'react-native-multiple-styles';
import Button from './../atoms/Button/Button';
import InputText from './../atoms/InputText/InputText';
import {bindActionCreators} from 'redux';
import * as counterActions from '../../actions/index';
import { connect } from 'react-redux';
const Dimensions = require('Dimensions');

const viewport = {
  fullWidth: Dimensions.get('window').width,
  fullHeight: Dimensions.get('window').height,
};

class Login extends PureComponent {

  componentDidMount(){
    this.props.actions.actionCheck();
  }

  loginClick = () =>{
    console.log('Login click');
  }

  registerClick = () => {
    console.log('Register click')
  }

  username=(value)=>{
    console.log('username')
  }

  password=(value)=>{
    console.log('password')
  }

  render() {
    return (
      <View style={styles.container}>

        <InputText
          placeholder="Enter the username"
          inputFieldValue={this.username}
        />

        <InputText
          placeholder="Enter the password"
          secureTextEntry={true}
          inputFieldValue={this.password}
        />

        <Button
          onClick={this.loginClick}
        >
          LOGIN
        </Button>
        <Button
          onClick={this.registerClick}
        >
          REGISTER
        </Button>
      </View>
    );
  }
}

export default connect(state => ({
    count : state.freiraum,
  }),
  (dispatch) => ({
    actions: bindActionCreators(counterActions, dispatch)
  })
)(Login);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button:{
    display: 'flex',
    backgroundColor:'#000000',
    width: viewport.fullWidth* 0.80,
    padding: viewport.fullWidth * 0.03,
    alignItems:'center',
    justifyContent:'center',
  },
  buttonText:{
    color:'#fff',
    fontSize:20,
  }
});