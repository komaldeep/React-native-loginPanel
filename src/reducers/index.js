import freiraum from './reducers';

export {
  freiraum
};
