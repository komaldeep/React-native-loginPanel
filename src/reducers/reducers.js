const initialState = {
  count: false,
};

export default function freiraum(state = initialState, action = {}) {
  switch (action.type) {
    case "COUNT":
      return {
        ...state,
        count: action.payload
      }

    default:
      return state;
  }
}
