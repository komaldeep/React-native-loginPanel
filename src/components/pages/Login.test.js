import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import Login from './Login';
import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import * as reducers from '../../reducers/index';

const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
const reducer = combineReducers(reducers);
const store = createStoreWithMiddleware(reducer);

test('renders without crashing', () => {
  const rendered = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('on click of button', () => {
  const component = shallow(<Provider store={store}><Login /></Provider>);
  component.simulate( 'onPress' );
  expect(component).toMatchSnapshot();
});