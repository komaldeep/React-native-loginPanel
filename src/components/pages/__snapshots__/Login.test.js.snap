// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on click of button 1`] = `<Connect(Login) />`;

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter the username"
    secureTextEntry={false}
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
        "height": 40,
        "marginBottom": 10,
        "width": 600,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter the password"
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
        "height": 40,
        "marginBottom": 10,
        "width": 600,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 22.5,
        "width": 600,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      LOGIN
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "padding": 22.5,
        "width": 600,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      REGISTER
    </Text>
  </View>
</View>
`;
